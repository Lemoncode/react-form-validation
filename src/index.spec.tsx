import * as React from 'react';
import { shallow } from 'enzyme';
import {
  createFormValidation,
  ValidationConstraints,
  Validators,
} from 'lc-form-validation';
import { useValidation } from './index';

describe('useValidation specs', () => {
  describe('entity', () => {
    it('should render component with initial entity value', () => {
      // Arrange
      const initialEntity = {
        test: 'test prop',
      };

      const constraints: ValidationConstraints = {
        fields: {
          name: [{ validator: Validators.required }],
        },
      };
      const validation = createFormValidation(constraints);
      const TestComponent = () => {
        const [entity] = useValidation(validation, initialEntity, null);
        return <div>{entity.test}</div>;
      };

      // Act
      const component = shallow(<TestComponent />);

      // Assert
      expect(component).toMatchSnapshot();
    });
  });
  describe('entity error', () => {
    it('should render component with initial entity error value', () => {
      // Arrange
      const initialEntity = {
        test: 'test prop',
      };

      const initialEntityError = {
        testError: 'testError prop',
      };

      const constraints: ValidationConstraints = {
        fields: {
          name: [{ validator: Validators.required }],
        },
      };
      const validation = createFormValidation(constraints);
      const TestComponent = () => {
        const [entity, entityError] = useValidation(
          validation,
          initialEntity,
          initialEntityError
        );
        return <div>{entityError.testError}</div>;
      };

      // Act
      const component = shallow(<TestComponent />);

      // Assert
      expect(component).toMatchSnapshot();
    });
  });
});
