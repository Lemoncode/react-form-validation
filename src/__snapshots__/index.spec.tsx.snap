// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useValidation specs entity error should render component with initial entity error value 1`] = `
<div>
  testError prop
</div>
`;

exports[`useValidation specs entity should render component with initial entity value 1`] = `
<div>
  test prop
</div>
`;
